tests/phpunit/integration/Modules/Analytics_4/Synchronize_PropertyTest.php

```diff
@@ -0,0 +1,225 @@
+		wp_set_current_user( $user->ID );
+
+		$context              = new Context( GOOGLESITEKIT_PLUGIN_MAIN_FILE );
+		$this->options        = new Options( $context );
+		$this->user_options   = new User_Options( $context );
+		$this->authentication = new Authentication( $context, $this->options, $this->user_options );
+		$modules              = new Modules( $context, $this->options, $this->user_options, $this->authentication );
+		$dismissed_items      = new Dismissed_Items( $this->user_options );
+		$encrypted_options    = new Encrypted_Options( $this->options );
+		$permissions          = new Permissions( $context, $this->authentication, $modules, $this->user_options, $dismissed_items );
+
+		// Fake a valid authentication token on the client.
+		$this->authentication->get_oauth_client()->set_token( array( 'access_token' => 'valid-auth-token' ) );
+		$this->authentication->verification()->set( true );
+
+		$encrypted_options->set(
+			Credentials::OPTION,
+			array(
+				'oauth2_client_id'     => 'test-client-id',
+				'oauth2_client_secret' => 'test-client-secret',
+			)
+		);
+		add_filter( 'googlesitekit_setup_complete', '__return_true', 100 );
+
+		$permissions->register();
+
+		$this->analytics_4 = new Analytics_4(
+			$context,
+			$this->options,
+			$this->user_options,
+			$this->authentication
+		);
+
+		$this->set_analytics_settings();
+
+		$this->synchronize_property = new Synchronize_Property(
+			$this->analytics_4,
+			$this->user_options,
+			$this->authentication
+		);
+	}
+
+	public function set_analytics_settings() {
+		// Ensure Anlytics 4 module is connected
+		$this->options->set(
+			Analytics_4_Settings::OPTION,
+			array(
+				'accountID'       => '12345678',
+				'propertyID'      => '987654321',
+				'webDataStreamID' => '1234567890',
+				'measurementID'   => 'G-A1B2C3D4E5',
+			)
+		);
+
+		// Set owner ID, using base analytics settings.
+		$this->options->set(
+			Settings::OPTION,
+			array(
+				'ownerID' => get_current_user_id(),
+			)
+		);
+	}
+
+	public function fake_analytics_4_property_object( $property_id, $create_time ) {
+		FakeHttp::fake_google_http_handler(
+			$this->analytics_4->get_client(),
+			function ( Request $request ) use ( $property_id, $create_time ) {
+				$url = parse_url( $request->getUri() );
+
+				$property = new GoogleAnalyticsAdminV1betaProperty();
+				$property->setCreateTime( $create_time );
+				$property->setName( "properties/{$property_id}" );
+
+				if ( "/v1beta/properties/{$property_id}" === $url['path'] ) {
+					return new Response(
+						200,
+						array(),
+						json_encode(
+							$property
+						)
+					);
+				}
+
+				return new Response( 200 );
+			}
+		);
+	}
+
+	public function test_register() {
+		remove_all_actions( Synchronize_Property::CRON_SYNCHRONIZE_PROPERTY );
+
+		$this->synchronize_property->register();
+		$this->synchronize_property->maybe_schedule_synchronize_property();
```

Comment:
`maybe_schedule_synchronize_property` shouldn't be part of this test as it's specific to the `register` method. Instead it should be tested in its own case, i.e. `test_maybe_schedule_synchronize_property` where we'd want to test the conditions there.